python/sglang/srt/managers/scheduler.py

```diff
@@ -491,7 +507,40 @@ def recv_requests(self) -> List[Req]:
         else:
             recv_reqs = None
 
-        if self.tp_size != 1 and not self.server_args.enable_dp_attention:
+        if self.server_args.enable_dp_attention:
+            if self.attn_tp_rank == 0:
+                work_reqs = [
+                    req
+                    for req in recv_reqs
+                    if isinstance(
+                        req, (TokenizedGenerateReqInput, TokenizedEmbeddingReqInput)
+                    )
+                ]
+                control_reqs = [
+                    req
+                    for req in recv_reqs
+                    if not isinstance(
+                        req, (TokenizedGenerateReqInput, TokenizedEmbeddingReqInput)
+                    )
+                ]
+            else:
+                work_reqs = None
+                control_reqs = None
+
+            if self.attn_tp_size != 1:
+                attn_tp_rank_0 = self.dp_rank * self.attn_tp_size
+                work_reqs = broadcast_pyobj(
```

Comment:
Does this mean the embedding can only have same `attn_tp_size` instead of `tp_size` for sharding its parameters? If the embedding is sharded on the whole tp group, but here we just have inputs ready for a small group, the word vector we computed will be a part of whole tensor?